homeassistant/components/input_boolean/__init__.py

```diff
@@ -52,20 +83,57 @@ def is_on(hass, entity_id):
     return hass.states.is_state(entity_id, STATE_ON)
 
 
-async def async_setup(hass, config):
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
     """Set up an input boolean."""
     component = EntityComponent(_LOGGER, DOMAIN, hass)
+    id_manager = collection.IDManager()
+
+    yaml_collection = collection.YamlCollection(
+        logging.getLogger(f"{__name__}.yaml_collection"), id_manager
+    )
+    collection.attach_entity_component_collection(
+        component, yaml_collection, lambda conf: InputBoolean(conf, from_yaml=True)
+    )
+
+    storage_collection = InputBooleanStorageCollection(
+        Store(hass, STORAGE_VERSION, STORAGE_KEY),
+        logging.getLogger(f"{__name__}_storage_collection"),
+        id_manager,
+    )
+    collection.attach_entity_component_collection(
+        component, storage_collection, InputBoolean
+    )
+
+    await yaml_collection.async_load(
+        [{CONF_ID: id_, **(conf or {})} for id_, conf in config[DOMAIN].items()]
+    )
+    await storage_collection.async_load()
 
-    entities = await _async_process_config(config)
+    collection.StorageCollectionWebsocket(
+        storage_collection, DOMAIN, DOMAIN, CREATE_FIELDS, UPDATE_FIELDS
+    ).async_setup(hass)
 
-    async def reload_service_handler(service_call):
+    async def _collection_changed(
```

Comment:
I wonder if we should add this to the collection helper in a future PR. We're going to add it everywhere ?

Reply:
+1 for having this as a helper.
I also wouldn't mind if the `Child` class attach the `_collection_changed` method automatically if one exists.

Reply:
well it's one or the other 👍

Reply:
I just was thinking about a standard `clean up entity registry on removals` listener which we could attach to `YamlCollection` or `StorageCollection` e.g. `yaml_collection.add_listener(collection.EntityRegistryCleanup)`